README.md

```
@@ -1,6 +1,228 @@
+- **Location**: `.git/hooks/update`
+- **Functionality**:
+  - Prevents direct pushes to protected branches (main/master)
+  - Requires pull requests for changes to protected branches
+  - Verifies merge status before allowing updates
+- **Implementation**: JavaScript (Node.js)
+
+### Required .gitignore Patterns
+
+The following patterns are enforced in `.gitignore`:
+
+```
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified
null

(MD040, fenced-code-language)